EKF/control.cpp

```diff
@@ -720,9 +730,10 @@ void Ekf::controlHeightFusion()
 				}
 			}
 
-		} else if (_baro_data_ready && !_baro_hgt_faulty && !_in_range_aid_mode) {
+		} else if (_baro_data_ready && !_baro_hgt_faulty && _rng_hgt_faulty) {
```

Comment:
I don't understand why a check for the range height is needed here.

E.g. what happens if not in range aid mode but the ranger is good (`_rng_hgt_faulty == false`)? It seems like then it would end up not fusing anything?